app/Http/Controllers/CategoryController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use App\Category;
+
+class CategoryController extends Controller
+{
+    public function open_category_admin()
+    {
+        $categories = Category::get();
+        // dd(count($user));
+        if (count($categories) == 0) {
+            return redirect(route('add_category_admin'));
+        } else {
+            return view('admin.admin_categories', compact('categories'));
+        }
+    }
+    public function add_category_admin()
+    {
+        return view('admin.add_category');
+    }
+    public function store_category_admin(Request $request)
+    {
+        $category = new Category;
```

Comment:
You can make this short by using the `::create` function in Laravel.

Please read about thesethis:
https://laravel.com/docs/5.7/eloquent#mass-assignment